pkg/asset/installconfig/aws/validation.go

```diff
@@ -51,7 +66,7 @@ func validatePlatform(ctx context.Context, meta *Metadata, fldPath *field.Path,
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceEndpoints"), platform.ServiceEndpoints, err.Error()))
 	}
 	if platform.DefaultMachinePlatform != nil {
-		allErrs = append(allErrs, validateMachinePool(ctx, meta, fldPath.Child("defaultMachinePlatform"), platform, platform.DefaultMachinePlatform)...)
+		allErrs = append(allErrs, validateMachinePool(ctx, meta, fldPath.Child("defaultMachinePlatform"), platform, platform.DefaultMachinePlatform, computeReq)...)
```

Comment:
For `DefaultMachinePlatform` I was thinking this was compute machine pools.

Reply:
so currently the installer uses 3 places to find the instance size,
- internal default for the region
- platform.defaultMachinePool
- corresponding machinePool

So if default machine pool was set to compute requirements and no control plane machinepool was set, we will allow the user to use the instance type only sufficient for compute.

I think setting it to control plane requirements (which is larger) means that it will satisfy our requirement more accurately.

Reply:
Updated to compare defaultMachinePool against control plane requirements.